vispy/scene/cameras/base_camera.py

```diff
@@ -465,13 +475,55 @@ def _set_scene_transform(self, tr):
 
-        # Apply same state to linked cameras, but prevent that camera
-        # to return the favor
-        for cam in self._linked_cameras:
-            if cam is self._linked_cameras_no_update:
-                continue
-            try:
-                cam._linked_cameras_no_update = self
-                cam.set_state(self.get_state())
-            finally:
-                cam._linked_cameras_no_update = None
+        
+        if len(self._linked_cameras) != 0:
+            if isinstance(self._linked_cameras[0], list):
+                ## link only one axis between to cameras
+                for i in range(len(self._linked_cameras)):
+                    cam = self._linked_cameras[i][0]
+                    self_state = self.get_state()
+
+                    self.last_self = self.get_state()
+                    self.last_cam = cam.get_state()
+
+                    top = self.last_self["rect"].top
+                    bottom = self.last_self["rect"].bottom
+                    left = self.last_self["rect"].left
+                    right = self.last_self["rect"].right
+
+                    if (self._linked_cameras[0][1] == "x"):
+                        self_state["rect"].pos = (self_state["rect"].pos[0], self.last_cam["rect"].pos[1])
+                        self_state["rect"].top = self.last_cam["rect"].top
+                        self_state["rect"].bottom = self.last_cam["rect"].bottom
+                    elif (self._linked_cameras[0][1] == "y"):
+                        self_state["rect"].pos = (self.last_cam["rect"].pos[0], self_state["rect"].pos[1])
+                        self_state["rect"].left = self.last_cam["rect"].left
+                        self_state["rect"].right = self.last_cam["rect"].right
+                    else:
+                        raise ValueError(f"Axis align has to be either 'x' or 'y' not {self._linked_cameras[0][1]}")
+
+                    cam.set_state(self_state)
+
+                    if (self._linked_cameras[0][1] == "x"):
+                        self_state["rect"].pos = (self_state["rect"].pos[0], self.last_self["rect"].pos[1])
+                        self_state["rect"].top = top
+                        self_state["rect"].bottom = bottom
+                    elif (self._linked_cameras[0][1] == "y"):
+                        self_state["rect"].pos = (self.last_cam["rect"].pos[0], self_state["rect"].pos[1])
+                        self_state["rect"].left = left
+                        self_state["rect"].right = right
+                    else:
+                        raise ValueError(f"Axis align has to be either 'x' or 'y' not {self._linked_cameras[0][1]}")
+
+                    self.set_state(self_state)
+            else:
+                # Apply same state to linked cameras, but prevent that camera
+                # to return the favor
+                for i in range(len(self._linked_cameras)):
+                    if self._linked_cameras[i] is self._linked_cameras_no_update:
```

Comment:
It seems like this `else` block's logic (all of this `linked_cameras_no_update` stuff) needs to be incorporated into the above `if` block otherwise the state setting above will cause a loop between multiple cameras. Right?